tests/test_pyav.py

```diff
@@ -0,0 +1,329 @@
+        stream.height = frames.shape[1]
+
+        for frame in frames:
+            av_frame = av.VideoFrame.from_ndarray(frame, format="rgb24")
+            for packet in stream.encode(av_frame):
+                container.mux(packet)
+
+        for packet in stream.encode():
+            container.mux(packet)
+
+    mp4_bytes = iio.imwrite(
+        "<bytes>",
+        frames,
+        format_hint=".mp4",
+        plugin="pyav",
+        codec="libx264",
+    )
+
+    expected = Path(tmp_path / "expected.mp4").read_bytes()
+    # actual = Path(tmp_path / "actual.mp4").read_bytes()
+    assert expected == mp4_bytes
+
+
+def test_metadata(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        meta = plugin.metadata()
+        assert meta["profile"] == "High 4:4:4 Predictive"
+        assert meta["codec"] == "h264"
+
+        plugin.metadata(index=4)
+    print("")
+
+
+def test_properties(test_images: Path):
+    with iio.imopen(str(test_images / "cockatoo.mp4"), "r", plugin="pyav") as plugin:
+        with pytest.raises(IOError):
+            # subsampled format
+            plugin.properties(format="yuv420p")
+
+        with pytest.raises(IOError):
+            # components per channel differs
+            plugin.properties(format="nv24")
+
+        props = plugin.properties()
+        assert props.shape == (3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is False
+
+        props = plugin.properties(index=4, format="rgb48")
+        assert props.shape == (720, 1280, 3)
+        assert props.dtype == np.uint16
+        assert props.is_batch is False
+
+        props = plugin.properties(index=None)
+        assert props.shape == (280, 3, 720, 1280)
+        assert props.dtype == np.uint8
+        assert props.is_batch is True
+
+
+def test_video_format_to_dtype():
+    fake_formats = [
+        "vdpau",
+        "dxva2_vld",
+        "vaapi_moco",
+        "drm_prime",
+        "mediacodec",
+        "qsv",
+        "mmal",
+        "vulkan",
+        "vaapi_vld",
+        "d3d11",
+        "d3d11va_vld",
+        "videotoolbox_vld",
+        "vaapi_idct",
+        "opencl",
+        "cuda",
+        "xvmc",
+    ]
+
+    for name in video_format_names:
+        format = av.VideoFormat(name)
+
+        if name in fake_formats:
+            # should fail
+            with pytest.raises(ValueError):
+                _format_to_dtype(format)
+        else:
+            # should succseed
+            _format_to_dtype(format)
+
+
+def test_filter_sequence(test_images):
+    frames = iio.imread(
+        test_images / "cockatoo.mp4",
+        index=None,
+        plugin="pyav",
+        filter_sequence=[
+            ("drawgrid", "w=iw/3:h=ih/3:t=2:c=white@0.5"),
+            ("scale", {"size": "vga", "flags": "lanczos"}),
+            ("tpad", "start=3"),
+        ],
+    )
+
+    assert frames.shape == (283, 3, 480, 640)
```

Comment:
Do images really come out as `time, color, height, width`? Typically you would think of them as coming as `time, height, width, color`

Reply:
@hmaarrfk Yes, because we don't modify the raw pixel format unless told to do so. 

`cockatoo.mp4` uses `yuv444p`, which is a planar format and it will thus come out as `time, channel, height, width`. Though, we could easily make it something more familiar by passing `format="rgb24"` which would then produce `time, height, width, channel`.

Reply:
@hmaarrfk Yes, because we don't modify the raw pixel format unless told to do so. 

`cockatoo.mp4` uses `yuv444p`, which is a planar format and it will thus come out as `time, channel, height, width`. Though, we could easily make it something more familiar by passing `format="rgb24"` which would then produce `time, height, width, channel`.

Reply:
Regardless, the final image provided to the user would be `height, width, color`, right?

Reply:
Oh, it does not. IMO imageio should make sure that it is.

Reply:
> Oh, it does not. IMO imageio should make sure that it is.

Yes, `p` indicates planar (channel-first) formats. Well technically not channel-first, because you have weird formats like `nv24` which are planar and mix channel-first and channel-last, but yes. Usually, it means channel-first :D

Regarding the discussion of making sure that frames will be channel-last, I think we should continue that discussion in one place (here: https://github.com/imageio/imageio/issues/504)